### PR TITLE
Soc/xtensa: Convert to use DEVICE_DT_GET 

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

File: soc/xtensa/intel_adsp/common/irq.c [hunk 1]
```diff
@@ -106,25 +106,25 @@ int z_soc_irq_is_enabled(unsigned int irq)
 
 	switch (XTENSA_IRQ_NUMBER(irq)) {
 	case DT_IRQN(CAVS_INTC_NODE(0)):
-		dev_cavs = device_get_binding(DT_LABEL(CAVS_INTC_NODE(0)));
+		dev_cavs = DEVICE_DT_GET(CAVS_INTC_NODE(0));
 		break;
 	case DT_IRQN(CAVS_INTC_NODE(1)):
-		dev_cavs = device_get_binding(DT_LABEL(CAVS_INTC_NODE(1)));
+		dev_cavs = DEVICE_DT_GET(CAVS_INTC_NODE(0));
```
Comment (pdgendt) [inline soc/xtensa/intel_adsp/common/irq.c:112]:
```suggestion
		dev_cavs = DEVICE_DT_GET(CAVS_INTC_NODE(1));
```

File: soc/xtensa/intel_adsp/common/irq.c [hunk 2]
```diff
@@ -106,25 +106,25 @@ int z_soc_irq_is_enabled(unsigned int irq)
...
 		break;
 	case DT_IRQN(CAVS_INTC_NODE(2)):
-		dev_cavs = device_get_binding(DT_LABEL(CAVS_INTC_NODE(2)));
+		dev_cavs = DEVICE_DT_GET(CAVS_INTC_NODE(0));
```
Comment (pdgendt) [inline soc/xtensa/intel_adsp/common/irq.c:115]:
```suggestion
		dev_cavs = DEVICE_DT_GET(CAVS_INTC_NODE(2));
```

File: soc/xtensa/intel_adsp/common/irq.c [hunk 3]
```diff
@@ -106,25 +106,25 @@ int z_soc_irq_is_enabled(unsigned int irq)
...
 		break;
 	case DT_IRQN(CAVS_INTC_NODE(3)):
-		dev_cavs = device_get_binding(DT_LABEL(CAVS_INTC_NODE(3)));
+		dev_cavs = DEVICE_DT_GET(CAVS_INTC_NODE(0));
```
Comment (pdgendt) [inline soc/xtensa/intel_adsp/common/irq.c:118]:
```suggestion
		dev_cavs = DEVICE_DT_GET(CAVS_INTC_NODE(3));
```

**Review by pdgendt — CHANGES_REQUESTED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
> While this looks harmless would it be possible to wait for the SOF CI to be back in order before merging this? Thx!
> 
>     * [Revert "dma/cavs_hda: add missing hda link files to CMakeLists" #47524](https://github.com/zephyrproject-rtos/zephyr/pull/47524)

Looks like that was closed, merging.

**Comment by marc-hb** [comment]
> Looks like that was closed, merging.

SOF boot failures are not completely resolved yet but this change does not seem to make any difference.

**Pull request merged**